File: ts/src/ThesmsworksError.ts

import { Context } from './Context'


class ThesmsworksError extends Error {

  isThesmsworksError = true

  sdk = 'Thesmsworks'

  code: string
  ctx: Context

  constructor(code: string, msg: string, ctx: Context) {
    super(msg)
    this.code = code
    this.ctx = ctx
  }

}

export {
  ThesmsworksError
}

File: dart/lib/ThesmsworksError.dart
class ThesmsworksError extends Error {
  final bool isThesmsworksError = true;

  final String sdk = 'Thesmsworks';

  String code;
  String message;
  dynamic ctx;

  // Populated by makeError with the (cleaned) result and spec.
  dynamic result;
  dynamic spec;

  ThesmsworksError(this.code, this.message, [this.ctx]);

  @override
  String toString() => 'ThesmsworksError: ' + code + ': ' + message;
}
